app/src/main/java/com/orlove101/android/mvvmstoragetask/ui/cats/CatsViewModel.kt

```diff
@@ -0,0 +1,119 @@
+package com.orlove101.android.mvvmstoragetask.ui.cats
+
+import android.content.Context
+import androidx.lifecycle.*
+import com.orlove101.android.mvvmstoragetask.data.models.Cat
+import com.orlove101.android.mvvmstoragetask.persistence.Room.CatsDao
+import com.orlove101.android.mvvmstoragetask.persistence.Room.CurrentDatabase
+import com.orlove101.android.mvvmstoragetask.persistence.Room.PreferencesManager
+import com.orlove101.android.mvvmstoragetask.persistence.Room.SortOrder
+import com.orlove101.android.mvvmstoragetask.persistence.SQLite.CatsDatabaseHelper
+import com.orlove101.android.mvvmstoragetask.ui.ADD_TASK_RESULT_OK
+import com.orlove101.android.mvvmstoragetask.ui.EDIT_TASK_RESULT_OK
+import dagger.hilt.android.lifecycle.HiltViewModel
+import dagger.hilt.android.qualifiers.ApplicationContext
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.flatMapLatest
+import kotlinx.coroutines.flow.receiveAsFlow
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class CatsViewModel @Inject constructor(
+    private val catsDao: CatsDao,
+    private val preferencesManager: PreferencesManager,
+    private val state: SavedStateHandle,
+    private val catsDbHelper: CatsDatabaseHelper
+): ViewModel() {
+    val searchQuery = state.getLiveData("searchQuery", "")
+    val preferencesFlow = preferencesManager.preferencesFlow
+
+    private val catsEventChannel = Channel<CatsEvent>()
+    val catsEvent = catsEventChannel.receiveAsFlow()
```

Comment:
Please, use clear, understandable names. 
Cats Event - means nothing. Also you uses flow here, which is stream by mature. It is not a single event!